### PR TITLE
VEP calibration session data

### PR DESCRIPTION
**Pull request by lawhead**
# Overview

Added session.json output for VEP Calibration task to capture data needed for creating a model.

## Ticket

https://www.pivotaltracker.com/story/show/186522989

## Contributions

- Updated task data module to allow for task-specific Inquiry and Session data
- Added properties to display elements useful for recording in the session data.
- Added functionality to VEP Calibration task for outputting session data. General task data includes information about each box (coordinates, colors, flicker rate). Each Inquiry has information about which stimuli were presented in each box, which box contained the target symbol, as well as the flicker rate of that box.

## Test

- Ran all unit tests
- Ran a VEP calibration task and inspected the resulting session.json file.

### CODE REVIEW

File: bcipy/task/paradigm/vep/calibration.py [hunk 1]
```diff
@@ -226,14 +302,15 @@ def init_calibration_display(parameters: Parameters,
...
-                                      stim_height=0.1,
-                                      timing=timing,
-                                      stim_color=colors,
-                                      inquiry=[],
-                                      stim_length=1,
-                                      animation_seconds=parameters['time_vep_animation'])
+    stim_props = VEPStimuliProperties(
+        stim_font=parameters['font'],
+        stim_pos=box_config.positions,
+        stim_height=0.1,
```
Comment (tab-cmd) [inline bcipy/task/paradigm/vep/calibration.py:306]: This may be a good class variable so we can change it with task parameters as needed

**Review by tab-cmd — APPROVED** [review]
This looks good to me! I sent the output to @celikbasak to check it had all the information needed for our models.

Here are my notes for things to address in future PRs:
- I wasn't able to get the time spent output in the session.json on my machine. 
-  I encountered a weird double animation during inquiries. I'll send you a screen capture on Discord (too large for GitHub). 
- Fake data capability. I ended up hardcoding my frame rate and hacking in the fake parameter from start_task. We might consider adding that feature to help with development. This may be related to the double animation I experienced though!

**Comment by lawhead** [comment]
The double animation happens if the frame rate is incorrect. There is a VEP Display parameter for manually setting the frame rate, which overrides the psychopy computed value, which doesn't seem to work on my M2 Mac. You can try setting this value and see if that fixes it for you. I had to set this to 120hz even though I have my Display settings set to 60hz. I set this in the VEP calibration module, but I agree that exposing this as a parameter might be useful.

I'm not sure I'm computing the time_spent. I'll double-check that and add it to this PR.

**Comment by tab-cmd** [comment]
> The double animation happens if the frame rate is incorrect. There is a VEP Display parameter for manually setting the frame rate, which overrides the psychopy computed value, which doesn't seem to work on my M2 Mac. You can try setting this value and see if that fixes it for you. I had to set this to 120hz even though I have my Display settings set to 60hz. I set this in the VEP calibration module, but I agree that exposing this as a parameter might be useful.
> 
> I'm not sure I'm computing the time_spent. I'll double-check that and add it to this PR.

Okay! @celikbasak said this should be enough to train VEP models. We may find better ways to organize/make this available online, but I think it's worth waiting to test out / encounter pain points.

**Pull request merged**